src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+|-----------------|-------------------------------------|
+| \n              | linefeed (ASCII code 10)            |
+| \r              | carriage return (ASCII code 13)     |
+| \t              | tab (ASCII code 8)                  |
+| \\              | backslash                           |
+| \'              | single quote                        |
+| \"              | double quote                        |
+| \xNN            | two byte hexadecimal character code |
+| \u{NNNNNNN}     | unicode code sequence               |
+
+#### Example String Literals
+
+```
+"simple text"
+"two\nlines with \"escapes\""_varchar<80>
+"abcde"_fixedchar<5>
+"unicode chars: \xA9 \u{023B}"
+```
+
+### Binary Literals
+
+Binary literals are represented as strings containing hexadecimal digits (no escapes required).  Fixed binary literals
+are measured in bytes and thus have two hexadecimal characters per required byte.  A UUID is effectively a
+fixedbinary\<16> that ignores any provided dashes in its representation. 
+
+#### Example Binary Literals
+
+```
+"0123456789abcde"_binary
+"1234"_fixedbinary<2>
+"00-ABCD-1234"_uuid
+```
+
+### Time Literals
+
+There are two kinds of time literals:  literals that represent particular points in time and literals that represent
+durations.  The timestamps represent dates from the beginning of the year 1000 to the end of the year 9999.  The 
+interval_year duration has its precision in months and the interval_day duration has its precision in microseconds.
+
+To make the structured data easier to read the pseudo types ```year```, ```month```, ```day```, ```second```, and
+```microsecond``` (plural or singular) are suggested for readability.
+
+#### Example Time Literals
+
+```
+"1000-01-01 00:00:00.000000"_timestamp
+"1000-01-01 00:00:00.000000 UTC"_timestamp_tz
+"2020-12-20"_date
+"13:21"_time
+"13:21:12.012345"_time
+
+{5_years, 1_month}_interval_year
+{5_year, 1_months}_interval_year
+{4_days, 1_second, 13_microseconds}_interval_day
+```
+
+### Structured Literals
+
+Structure literals use the brace syntax.  Types are not necessary within the value section as the types should be 
+homogeneous if the structure is well-formed. 
+
+|                                                           | Description                                   |
+|-----------------------------------------------------------|-----------------------------------------------|
+| {}_list\<string>                                          | empty list of string                          |
+| {}_list\<string?>                                         | empty list of optional strings                |
+| {null, "a", "b"}_list\<string?>                           | list of optional strings                      |
+| {}_map\<int, string>                                      | empty map with integer keys and string values |
+| {42->"life", 32->"everything"}_map\<int, string>          | map with integer keys and string values       |
```

Comment:
Hmm...why `->` instead of `:` which is probably more universally expected?

Reply:
changed